### ISSUE TITLE
[ops]: Tensor placements are wrong after ttnn.all_reduce

### ISSUE THREAD

**Issue by ichovpanTT** [tenstorrent/tt-metal#50657]
### Component / Area

ttnn core

### Issue Type (optional)

None

### Observation 1

The tensor after ttnn.all_reduce still has a placement of Shard(dim=0) instead of Replicate, even though ttnn.all_reduce replicates tensors across devices.

### Observation 2

The tensor placements array isn't well defined right now since its length is based on the logical mesh shape, but the CCL ops are called on the physical axes. 

### Expected

The tensor placement should be modified to replicate after all_reduce

### 1. Steps (exact commands)

1. Run this python code:
```
import torch
import ttnn


def placements_str(topo):
    out = []
    for p in topo.placements():
        out.append(f"Shard(dim={p.dim})" if isinstance(p, ttnn.PlacementShard) else "Replicate")
    return out


def main():
    ttnn.set_fabric_config(ttnn.FabricConfig.FABRIC_1D)
    mesh = ttnn.open_mesh_device(ttnn.MeshShape(1, 2))
    try:
        torch.manual_seed(0)
        M, N = 32, 64

        mapper = ttnn.create_mesh_mapper(
            mesh,
            ttnn.MeshMapperConfig(
                placements=[ttnn.PlacementShard(0)],
                mesh_shape_override=ttnn.MeshShape([2]),
            ),
        )
        tt_in = ttnn.from_torch(
            torch.randn(2, M, N, dtype=torch.bfloat16),
            dtype=ttnn.bfloat16,
            layout=ttnn.TILE_LAYOUT,
            device=mesh,
            mesh_mapper=mapper,
        )
        print(f"BEFORE all_reduce: placements={placements_str(tt_in.tensor_topology())}")

        out = ttnn.all_reduce(tt_in, cluster_axis=1, num_links=1, topology=ttnn.Topology.Linear)
        ttnn.synchronize_device(mesh)

        print(f"AFTER  all_reduce: placements={placements_str(out.tensor_topology())}")
    finally:
        ttnn.close_mesh_device(mesh)
        ttnn.set_fabric_config(ttnn.FabricConfig.DISABLED)


if __name__ == "__main__":
    main()
```

2. Observe the outputs:
```
BEFORE all_reduce: placements=['Shard(dim=0)']
AFTER  all_reduce: placements=['Shard(dim=0)']
```

3. The AFTER outputs *should* be
```
AFTER  all_reduce: placements=['Replicate']
```

### 2. Input data / link or description

Any tensor sharded across devices that also has a tensor placement of Shard(any dim)

### 3. Frequency

Always

### 1. Software Versions

Ubuntu 22.04
Python: 3.10.12

### 2. Hardware Details

Any Blackhole/Wormhole system with >=2 devices available can be used

### Is this a regression?

Unknown

### Priority

None
